File: php/src/Model/TaskList.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

use Fivexer\SDK\Internal\Json;

/**
 * An unpaginated task listing — the parked and scheduled views return the whole set.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class TaskList
{
    public function __construct(
        /** @var list<Task> */
        public readonly array $tasks,
        public readonly int $count,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            tasks: Json::parseEach($data, 'tasks', [Task::class, 'fromArray']),
            count: (int) ($data['count'] ?? 0),
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/TaskList.java
package io.fivexer.sdk.model;

import java.util.List;

/**
 * An unpaginated task listing — the parked and scheduled views return the whole set.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class TaskList {
    private List<Task> tasks;
    private Integer count;

    public List<Task> getTasks() { return tasks; }
    public Integer getCount() { return count; }
}
